File: BlogServer/Blog.Model/Param/LinksParam.cs

using Blog.Model.Enum;

namespace Blog.Model.Param
{
    public class LinksInsertParam
    {
        public required string Name {  get; set; }
        public required string Avatar { get; set; }
        public required string Http_url { get; set; }
        public required StatusEnum Status { get; set; }
        public string? Description { get; set; }
    }

    public class LinksUpdateParam: LinksInsertParam
    { 
        public required int Id { get; set; }
    }
}

File: blog-server/src/module/admin/links/model/links.param.ts
import {IsString, IsInt, IsOptional} from 'class-validator'


export class LinksInsertParam {
    @IsString({ message: 'content必须是字符类型' })
    name: string

    @IsString({ message: 'content必须是字符类型' })
    avatar: string
    
    @IsString({ message: 'content必须是字符类型' })
    http_url: string

    @IsString()
    @IsOptional()
    description: string

    @IsInt()
    status?: number
}


export class LinksUpdateParam extends LinksInsertParam{
    @IsInt({ message: 'Id必须是整数类型' })
    id: number;
}
